pkg/kubelet/kubelet.go

```diff
@@ -2067,11 +2067,14 @@ func (kl *Kubelet) SyncTerminatingPod(_ context.Context, pod *v1.Pod, podStatus
 	// catch race conditions introduced by callers updating pod status out of order.
 	// TODO: have KillPod return the terminal status of stopped containers and write that into the
 	//  cache immediately
-	podStatus, err := kl.containerRuntime.GetPodStatus(ctx, pod.UID, pod.Name, pod.Namespace)
+	stoppedPodStatus, err := kl.containerRuntime.GetPodStatus(ctx, pod.UID, pod.Name, pod.Namespace)
 	if err != nil {
 		klog.ErrorS(err, "Unable to read pod status prior to final pod termination", "pod", klog.KObj(pod), "podUID", pod.UID)
 		return err
 	}
+	// Copy the IPs from the previous status, because containerRuntime.GetPodStatus
+	// only returns IPs for a running pod.
+	stoppedPodStatus.IPs = podStatus.IPs
```

Comment:
`podStatus` or `apiPodStatus` that is generated in  L2037?
`apiPodStatus := kl.generateAPIPodStatus(pod, podStatus, false)`

Reply:
Doesn't really matter, the generated status is taken from the `podStatus`. Only difference is it's sorted. I guess technically using the genertade version here would make the next conversion slightly faster, but I think it's probably better to use `podStatus` since the types are the same.

Reply:
Correct, we could copy the IP both from `podStatus` or from `apiPodStatus` (both contain the IP). 

My preference for copying IP from `podStatus` is that:
1. `apiPodStatus` is derived from `podStatus` in `generateAPIPodStatus`, so adjusting the underlying structure keeps them consistent, 
2. `apiPodStatus` has two fields `IP` and `IPs` which would need both be copied, so it would be slightly more code